### PR TITLE
fix: fix resources exceptions routes responses, add tests, update README

### PR DESCRIPTION
**Pull request by bolinocroustibat**
Fix several serious issues with resources exceptions routes:

- Fix serialisation of resource exception in API responses
- Added a proper Marshmallow schema for resource exception, so that it can be used for serialisation
- Allow posting of a new resources exception with no `tables_indexes`
- Added end-to-end API tests for all resources exceptions routes cases
- Remove resources exception routes to be excluded from auth, so that POST and DELETE are also authenticated routes
- Added documentation on how to make a curl request to the API to add a ressource exception

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -397,6 +397,13 @@ Refer to each section to learn how to launch them. The only differences from dev
 - use `HYDRA_SETTINGS` env var to point to your custom `config.toml`
 - use `HYDRA_APP_SOCKET_PATH` to configure where aiohttp should listen to a [reverse proxy connection (eg nginx)](https://docs.aiohttp.org/en/stable/deployment.html#nginx-configuration) and use `udata-hydra-app` to launch the app server
 
+## Adding large resources exceptions
+
+Here is an example of how to add a large resource exception to Hydra's database through the API:
+```bash
+curl -X POST https://dev-crawler.data.gouv.fr/api/resources-exceptions  -H "Authorization: Bearer abcdefghijklmopqrstuvw1234567890" -d "{'resource_id': 'f868cca6-8da1-4369-a78d-47463f19a9a3', 'table_indexes': {'SIRET': "index", "immatriculation": "index"}}"
```
Comment (maudetes) [inline README.md:404]: I think we can add a DELETE example as well

File: README.md [hunk 2]
```diff
@@ -397,6 +397,13 @@ Refer to each section to learn how to launch them. The only differences from dev
 - use `HYDRA_SETTINGS` env var to point to your custom `config.toml`
 - use `HYDRA_APP_SOCKET_PATH` to configure where aiohttp should listen to a [reverse proxy connection (eg nginx)](https://docs.aiohttp.org/en/stable/deployment.html#nginx-configuration) and use `udata-hydra-app` to launch the app server
 
+## Adding large resources exceptions
```
[the remark this answers is not in the document]
Comment (bolinocroustibat) [reply README.md:400]: Oh indeed, way better, I missed that in the doc. And by the way there are several updates to be done in that section, I pushed that as well. Thanks

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (bolinocroustibat) [reply README.md:404]: Yep, totally

Comment (bolinocroustibat) [reply README.md:404]: Yep, totally

File: udata_hydra/schemas/resource_exception.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+from marshmallow import Schema, fields
+
+
+class ResourceExceptionSchema(Schema):
+    id = fields.Str(required=True)
+    resource_id = fields.Str(required=True)
+    table_indexes = fields.Str(allow_none=True)
```
[the remark this answers is not in the document]
Comment (bolinocroustibat) [reply udata_hydra/schemas/resource_exception.py:7]: It's a dictionary/JSON, but it's received as a serialized string.

File: udata_hydra/routes/resources_exceptions.py [hunk 4]
```diff
@@ -1,57 +1,66 @@
...
     try:
         resource_exception: Record = await ResourceException.insert(
             resource_id=resource_id,
             table_indexes=table_indexes,
         )
     except ValueError as err:
-        raise web.HTTPBadRequest(text=f"Resource exception could not be created: {err}")
+        raise web.HTTPBadRequest(text=f"Resource exception could not be created: {str(err)}")
+    except UniqueViolationError:
+        raise web.HTTPBadRequest(text="Resource exception already exists")
```
[the remark this answers is not in the document]
Comment (bolinocroustibat) [reply udata_hydra/routes/resources_exceptions.py:50]: For now, yes, but I think we ought to create a PUT endpoint as well.
Let's do that in another PR.
Ticket here: https://github.com/datagouv/hydra/issues/177

**Pull request merged**